gensim/test/test_ldamodel_distributed.py

```diff
@@ -0,0 +1,527 @@
+    for testcase in testcases:
+        assert(isinstance(utils.get_random_state(testcase), np.random.RandomState))
+
+
+class TestLdaModel(unittest.TestCase, basetmtests.TestBaseTopicModel):
+    def setUp(self):
+        self.corpus = mmcorpus.MmCorpus(datapath('testcorpus.mm'))
+        self.class_ = ldamodel.LdaModel
+        # test non-standard port to see if it still locates the right nameserver
+        self.host = '127.0.0.1'
+        self.port = 3123
+        env = {
+            'PYRO_SERIALIZERS_ACCEPTED': 'pickle',
+            'PYRO_SERIALIZER': 'pickle',
+        }
+        self.ns_process = Subprocess.Popen([sys.executable, '-m', 'Pyro4.naming', '--host=' + host, '--port=' + str(port)], env=env)
+        self.worker_processes = [
+            Subprocess.Popen([sys.executable, '-m', 'gensim.models.lsi_worker', '--host', host, '--port', str(port)], env=env)
+            for _ in range(2)
+        ]
+        self.dispatcher_process = Subprocess.Popen([sys.executable, '-m', 'gensim.models.lsi_dispatcher', '--host', host, '--port', str(port)], env=env)
+        self.model = self.class_(corpus, id2word=dictionary, num_topics=2, passes=100, distributed=True, ns_conf=dict(host=self.host, port=self.port))
+
+    def testTransform(self):
+        passed = False
+        # sometimes, LDA training gets stuck at a local minimum
+        # in that case try re-training the model from scratch, hoping for a
+        # better random initialization
+        for i in range(25):  # restart at most 5 times
+            # create the transformation model
+            model = self.class_(id2word=dictionary, num_topics=2, passes=100)
+            model.update(self.corpus)
+
+            # transform one document
+            doc = list(corpus)[0]
+            transformed = model[doc]
+
+            vec = matutils.sparse2full(transformed, 2)  # convert to dense vector, for easier equality tests
+            expected = [0.13, 0.87]
+            # must contain the same values, up to re-ordering
+            passed = np.allclose(sorted(vec), sorted(expected), atol=1e-1)
+            if passed:
+                break
+            logging.warning(
+                "LDA failed to converge on attempt %i (got %s, expected %s)", i, sorted(vec), sorted(expected)
+            )
+        self.assertTrue(passed)
+
+    def testAlphaAuto(self):
+        model1 = self.class_(corpus, id2word=dictionary, alpha='symmetric', passes=10)
+        modelauto = self.class_(corpus, id2word=dictionary, alpha='auto', passes=10)
+
+        # did we learn something?
+        self.assertFalse(all(np.equal(model1.alpha, modelauto.alpha)))
+
+    def testAlpha(self):
+        kwargs = dict(
+            id2word=dictionary,
+            num_topics=2,
+            alpha=None
+        )
+        expected_shape = (2,)
+
+        # should not raise anything
+        self.class_(**kwargs)
+
+        kwargs['alpha'] = 'symmetric'
```

Comment:
Are these blocks independent tests?

If they are, then you should break this test method up into multiple methods, here and everywhere.